apps/arena/lib/arena/game/skill.ex

```diff
@@ -40,7 +41,28 @@ defmodule Arena.Game.Skill do
         Map.put(players_acc, player_id, target_player)
       end)
 
-    %{game_state | players: players}
+    # Crates
+
+    interactable_crates =
+      Crate.interactable_crates(game_state.crates)
+
+    crates =
+      Physics.check_collisions(circular_damage_area, interactable_crates)
+      |> Enum.reduce(game_state.crates, fn crate_id, crates_acc ->
+        real_damage = Player.calculate_real_damage(entity_player_owner, circle_hit.damage)
+
+        crate =
+          Map.get(crates_acc, crate_id)
+          |> Crate.take_damage(real_damage)
+
+        unless Crate.alive?(crate) do
+          send(self(), {:destroy_crate, crate_id})
```

Comment:
I understand why doing this, but how about no? Instead we can do nothing and in the game tick check for crates that died and mark them as destroyed or whatever we want

Reply:
hahaha i liked that way of rejecting something 😛  

sure thing! i moved this spawn power up handling to a handler in the main tick, also this makes mi wornders, what stops us to make this with the player? 🤔 

Also i added some sections to the main tick pipeline because it started to get a bit messy